File: app/src/main/java/com/karol/sezonnazdrowie/data/FoodItemDao.java
package com.karol.sezonnazdrowie.data;

import java.util.ArrayList;
import java.util.List;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

@Dao
public interface FoodItemDao {

    @Query("SELECT * FROM FoodItem WHERE mIsFruit = 1")
    List<FoodItem> getAllFruits();

    @Query("SELECT * FROM FoodItem WHERE mIsFruit = 0")
    List<FoodItem> getAllVegetables();

    @Insert
    void insertAll(List<FoodItem> items);

    @Query("SELECT * FROM FoodItem WHERE mName = :itemName")
    FoodItem getItem(String itemName);

    @Insert
    void insertAll(FoodItem[] items);
}

File: app/src/main/java/com/karol/sezonnazdrowie/data/FoodItemDao.kt
package com.karol.sezonnazdrowie.data

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query

@Dao
interface FoodItemDao {

    @get:Query("SELECT * FROM FoodItem WHERE isFruit = 1")
    val allFruits: List<FoodItem>

    @get:Query("SELECT * FROM FoodItem WHERE isFruit = 0")
    val allVegetables: List<FoodItem>

    @Insert
    fun insertAll(items: List<FoodItem>)

    @Query("SELECT * FROM FoodItem WHERE name = :itemName")
    fun getItem(itemName: String): FoodItem

    @Insert
    fun insertAll(items: Array<FoodItem>)
}
